crates/gitbutler-repo/src/repository_ext.rs

```diff
@@ -197,11 +196,28 @@ impl RepositoryExt for git2::Repository {
             } else if (status.is_index_deleted() && !status.is_wt_new()) || status.is_wt_deleted() {
                 tree_update_builder.remove(path);
             } else {
-                let file_path = worktree_path.join(path);
-                let file = std::fs::read(file_path)?;
-                let blob = self.blob(&file)?;
+                let file_path = worktree_path.join(path).to_owned();
 
-                tree_update_builder.upsert(path, blob, git2::FileMode::Blob);
+                if file_path.is_symlink() {
+                    let resolved_path = file_path.read_link()?;
+                    let path_str = resolved_path
+                        .to_str()
+                        .context("Failed to convert path to str")?;
+
+                    let blob = self.blob(path_str.as_bytes())?;
+                    tree_update_builder.upsert(path, blob, git2::FileMode::Link);
+                } else {
+                    let file = std::fs::read(&file_path)?;
```

Comment:
Instead of reading the file into memory and than hashing it, the hash can be computed [in a streaming manner](https://docs.rs/git2/latest/git2/struct.Repository.html#method.blob_path).